dsl/src/main/java/tech/pegasys/peeps/privacy/TransactionManager.java

```diff
@@ -0,0 +1,235 @@
+import org.testcontainers.containers.ContainerLaunchException;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.MountableFile;
+
+public abstract class TransactionManager implements NetworkMember {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final int CONTAINER_PEER_TO_PEER_PORT = 8080;
+  private static final int CONTAINER_HTTP_RPC_PORT = 8888;
+  private static final int ALIVE_STATUS_CODE = 200;
+  private static final String AM_I_ALIVE_ENDPOINT = "/upcheck";
+
+  protected final GenericContainer<?> container;
+  private final TransactionManagerRpc transactionManagerRpc;
+  private final TransactionManagerRpcExpectingData rpc;
+
+  // TODO stronger typing than String
+  private final String networkP2PAddress;
+  private final String networkRpcAddress;
+
+  // TODO typing for key?
+  protected final String id;
+
+  public TransactionManager(
+      final TransactionManagerConfiguration config, final GenericContainer<?> container) {
+    this.container = container;
+    this.networkP2PAddress =
+        String.format("http://%s:%s", config.getIpAddress().get(), CONTAINER_PEER_TO_PEER_PORT);
+    this.networkRpcAddress =
+        String.format("http://%s:%s", config.getIpAddress().get(), CONTAINER_HTTP_RPC_PORT);
+
+    // TODO just using the first key, selecting the identity could be an option for
+    // multi-key TransactionManager
+    this.id = ClasspathResources.read(config.getPublicKeys().get(0).get());
+    this.transactionManagerRpc = new TransactionManagerRpc(config.getVertx(), id, dockerLogs());
+    this.rpc = new TransactionManagerRpcExpectingData(transactionManagerRpc);
+  }
+
+  public void awaitConnectivity(final Collection<TransactionManager> collection) {
+    collection.parallelStream().forEach(this::awaitConnectivity);
+  }
+
+  @Override
+  public void start() {
+    try {
+      container.start();
+
+      container.followOutput(
+          outputFrame -> LOG.info("{}: {}", getNodeName(), outputFrame.getUtf8String()));
+
+      transactionManagerRpc.bind(
+          container.getContainerId(),
+          container.getContainerIpAddress(),
+          container.getMappedPort(CONTAINER_HTTP_RPC_PORT));
+
+      // TODO validate the node has the expected state, e.g. consensus, genesis,
+      // networkId,
+      // protocol(s), ports, listen address
+
+      logTransactionManagerDetails();
+      logPortMappings();
+      logContainerNetworkDetails();
+    } catch (final ContainerLaunchException e) {
+      LOG.error(container.getLogs());
+      throw e;
+    }
+  }
+
+  @Override
+  public void stop() {
+    if (container != null) {
+      container.stop();
+    }
+    if (transactionManagerRpc != null) {
+      transactionManagerRpc.close();
+    }
+  }
+
+  public abstract String getNodeName();
```

Comment:
nit: you can probably use `getClass().getSimpleName()` ... so each subclass would returns its own name.

Reply:
rather this become a user defined name for the node, rather than identifying it as tessera/orion - eg. "Node1_tessera", then reuse this string as a prefix in log msgs.